servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpCookiePair.java

```diff
@@ -87,7 +87,7 @@ public static HttpCookiePair parseCookiePair(final CharSequence sequence, int na
 
     static HttpCookiePair parseCookiePair0(final CharSequence sequence, int nameStart, int nameLength, int valueEnd) {
         final int valueStart = nameStart + nameLength + 1;
-        if (valueEnd - 1 < valueStart) {
+        if (valueEnd <= valueStart || valueStart < 0) {
```

Comment:
`valueStart` is always >= 0

Reply:
Ignore the previous comment for now. Only if we validate each input param independently, it will make sense.